typing_extensions/src_py3/typing_extensions.py

```diff
@@ -1595,3 +1604,93 @@ class Point2D(TypedDict):
     The class syntax is only supported in Python 3.6+, while two other
     syntax forms work for Python 2.7 and 3.2+
     """
+
+
+if HAVE_ANNOTATED:
+    class _Annotated(typing._GenericAlias, _root=True):
+        """Runtime representation of an annotated type.
+
+        At its core 'Annoted[t, dec1, dec2...]' is an alias for the type 't'
+        with extra annotations. The alias behaves like a normal typing alias,
+        instantiating is the same as instantiating the underlying type, binding
+        it to types is also the same...
+        """
+
+        def __init__(self, origin, extras):
+            if isinstance(origin, _Annotated):
+                extras = origin.__extras__ + extras
+                origin = origin.__origin__
+            super().__init__(origin, origin)
+            self.__extras__ = extras
+
+        def copy_with(self, params):
+            new_type, = params
+            return _Annotated(new_type, self.__extras__)
+
+        def __repr__(self):
+            return "typing_extensions.Annotated[{}, {}]".format(
+                typing._type_repr(self.__origin__),
+                ", ".join(repr(a) for a in self.__extras__)
+            )
+
+        def __reduce__(self):
+            return operator.getitem, (
+                Annotated, (self.__origin__,) + self.__extras__
+            )
+
+        def __eq__(self, other):
+            if not isinstance(other, _Annotated):
+                return NotImplemented
+            if self.__origin__ != other.__origin__:
+                return False
+            return self.__extras__ == other.__extras__
+
+        def __hash__(self):
+            return hash((self.__origin__, self.__extras__))
+
+
+    class Annotated:
```

Comment:
I would make `Annotated` a `_SpecialForm` on Python 3.7 (like `Optional`, `Any`, etc.), it doesn't need to be a class. But this is not important for the PoC implementation if it is tricky.

Reply:
_SpecialForm seems to be a of a no-go unless we change the standard library; it just special cases everything based on class name.